src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -92,6 +79,13 @@ internal sealed class SourceOrdinaryMethodSymbol : SourceOrdinaryMethodSymbolBas
         {
             Debug.Assert(diagnostics.DiagnosticBag is object);
 
+            // Compute the type parameters.  If empty (the common case), directly point at the singleton to reduce the
+            // amount of pointers-to-arrays this type needs to store.
+            var typeParameters = MakeTypeParameters(syntax, diagnostics);
```

Comment:
>var typeParameters = MakeTypeParameters(syntax, diagnostics); [](http://example.com/codeflow?start=12&length=61)

I think it would be better to initialize ```_typeParameterInfo``` lazily and not do this additional work in the constructor

Reply:
After looking at changes for ```SourceOrdinaryMethodSymbolBase```, I see that we used to do this in base constructor. Moving the work to the derived type constructor should be fine then.